pennylane/ops/op_math/decompositions/ross_selinger.py

```diff
@@ -79,7 +91,103 @@ def _domain_correction(theta: float) -> tuple[float, ZOmega]:
     return 0.0, ZOmega(d=1)  # -pi/4 <= |theta| < pi/4 / 7pi/4 <= |theta| < 8pi/4
 
 
-def rs_decomposition(op, epsilon, *, max_search_trials=20, max_factoring_trials=1000):
+def apply_clifford_from_idx(idx, wire):
+    """Apply a Clifford gate sequence by index on the specified wire.
+
+    This function maps an integer index to one of the standard Clifford sequences
+    defined in `clifford_keys_unwired`. The returned function uses `qml.cond`
+    to select and apply the correct sequence in QJIT-compatible form.
+
+    Args:
+        idx (int): Index into the Clifford sequence list.
+        wire (int): Target wire.
+
+    Returns:
+        Callable: A conditional function that applies the indexed Clifford sequence.
+    """
+    keys = _clifford_keys_unwired()
+
+    def make_fn(seq):
+        def fn():
+            for g in seq:
+                g(wire)
+
+        return fn
+
+    # Build conditional cases: for each index, associate it with a function
+    cases = [(idx == i, make_fn(seq)) for i, seq in enumerate(keys)]
+
+    # Extract the first condition
+    head_cond, head_fn = cases[0]
+
+    # Remaining cases handled as "elif"
+    elifs = cases[1:]
+
+    return qml.cond(head_cond, head_fn, elifs=elifs)
```

Comment:
Just thought of another todo
```suggestion
    # TODO: use qml.switch once available
    return qml.cond(head_cond, head_fn, elifs=elifs)
```